src/__tests__/role.js

```diff
@@ -24,12 +24,108 @@ Here are the available roles:
 test('logs error when there are no available roles', () => {
   const {getByRole} = render('<div />')
   expect(() => getByRole('article')).toThrowErrorMatchingInlineSnapshot(`
-"Unable to find an element with the role "article"
+"Unable to find an accessible  element with the role "article"
 
-There are no available roles.
+There are no accessible roles.
 
 <div>
   <div />
 </div>"
 `)
 })
+
+test('by default excludes elements that have the html hidden attribute or any of their parents', () => {
+  const {getByRole} = render('<div hidden><ul /></div>')
+
+  expect(() => getByRole('list')).toThrowErrorMatchingInlineSnapshot(`
+"Unable to find an accessible  element with the role "list"
+
+There are no accessible roles.
```

Comment:
We should probably add something to this message that says: "But there are some inaccessible roles. If you wish to access them, then set the hidden option to true. Learn more about this here: linktodocs.com"

Reply:
This would require a lot of refactoring if we want to actually determine if there are hidden roles.

Would it be sufficient to say "There **might** be inaccessible roles" instead?

Reply:
Extended the message and added a link to the docs but didn't implement displaying accessible vs. inaccessible roles.